pkg/controller/hvpa/hvpa_controller.go

```diff
@@ -951,47 +1089,139 @@ func getWeightedRequests(vpaStatus *vpa_api.VerticalPodAutoscalerStatus, hvpa *a
 				weightedCPU := currCPU
 
 				log.V(3).Info("VPA", "weighted target mem", weightedMem, "weighted target cpu", weightedCPU)
-				log.V(3).Info("VPA sclae down", "minimum CPU delta", scaleDownMinDeltaCPU.String(), "minimum memory delta", scaleDownMinDeltaMem, "scale down enabled", scaleDownEnaled)
+				log.V(3).Info("VPA sclae down", "minimum CPU delta", scaleDownMinDeltaCPU.String(), "minimum memory delta", scaleDownMinDeltaMem)
 				log.V(3).Info("VPA scale up", "minimum CPU delta", scaleUpMinDeltaCPU.String(), "minimum memory delta", scaleUpMinDeltaMem, "HPA condition ScalingLimited", hpaScaleOutLimited)
 
-				if hpaScaleOutLimited && diffMem.Sign() > 0 && diffMem.Cmp(*scaleUpMinDeltaMem) > 0 &&
-					(overrideScaleUpStabilization || lastScaleTimeDuration > scaleUpStabilizationWindow) {
-					// If the difference is greater than minimum delta, AND
-					// scale stabilization window is expired
-					log.V(2).Info("VPA", "Scaling up", "memory", "Container", container.Name)
-					newPodSpec.Containers[id].Resources.Requests[corev1.ResourceMemory] = weightedMem.DeepCopy()
-					resourceChange = true
-				} else if scaleDownEnaled && diffMem.Sign() < 0 && negDiffMem.Cmp(*scaleDownMinDeltaMem) > 0 &&
-					lastScaleTimeDuration > scaleDownStabilizationWindow {
-					log.V(2).Info("VPA", "Scaling down", "memory", "Container", container.Name)
-					newPodSpec.Containers[id].Resources.Requests[corev1.ResourceMemory] = weightedMem.DeepCopy()
-					resourceChange = true
+				if vpaWeight == 0 {
+					outTargetWeight[corev1.ResourceMemory] = rec.Target.Memory().DeepCopy()
+					blockedByWeight = true
+
+				} else if diffMem.Sign() > 0 {
+					if hpaScaleOutLimited == false || *hvpa.Spec.Vpa.UpdatePolicy.UpdateMode == autoscalingv1alpha1.UpdateModeOff {
+						outTargetUpdatePolicy[corev1.ResourceMemory] = rec.Target.Memory().DeepCopy()
+						blockedByUpdatePolicy = true
+
+					} else if overrideScaleUpStabilization == false && lastScaleTimeDuration < scaleUpStabilizationWindow {
+						outTargetStabilizationWindow[corev1.ResourceMemory] = rec.Target.Memory().DeepCopy()
+						blockedByStabilizationWindow = true
+
+					} else if diffMem.Cmp(*scaleUpMinDeltaMem) < 0 {
+						outTargetMinChanged[corev1.ResourceMemory] = rec.Target.Memory().DeepCopy()
+						blockedByMinChange = true
+
+					} else {
+						log.V(2).Info("VPA", "Scaling up", "memory", "Container", container.Name)
+						newPodSpec.Containers[id].Resources.Requests[corev1.ResourceMemory] = weightedMem.DeepCopy()
+						// Override VPA status in outVpaStatus with weighted value
+						outTarget[corev1.ResourceMemory] = weightedMem.DeepCopy()
+						resourceChange = true
```

Comment:
We should ideally still push to `outTargetWeight` `vpaMemTarget` if it is not equal to `weightedMem`. But we can pick this in a later issue.

Reply:
Resolving in favour of #20.